### ISSUE TITLE
Why ApolloStore#remove does not trigger changes to ApolloStoreWatcher?

### ISSUE THREAD

**Issue by piotr-pawlowski** [apollographql/apollo-android#2505]
Why ApolloStore#remove does not trigger changes to ApolloStoreWatcher? Should I publish an event manually with `publish` method?

**Comment by martinbonnin** [comment]
> Should I publish an event manually with publish method?

Yes.

To make it consistent with the `write` API, we should add a `removeAndPublish`

**Comment by piotr-pawlowski** [comment]
When are you gonna add this feature? :)

**Comment by martinbonnin** [comment]
Hi! It's unfortunately not as easy as I first imagined because removed cache keys would have to be collected through the whole deleted tree when `cascade` is true. Can you detail a bit more what your use case is? Maybe there's a more idiomatic solution than removing from the cache manually.

**Comment by piotr-pawlowski** [comment]
I'd like to remove some sensitive data (rows) from cache after they expire. I don't want to clear all cache but several rows.

**Comment by martinbonnin** [comment]
How do you handle expiration? Is it something from Apollo cache or do you handle this from "outside"?

**Comment by piotr-pawlowski** [comment]
I have my own solution for that. Each row has an expiration date and there is a background worker that clears expired rows.

BTW. What about #2532? Is is the same problem?

**Comment by martinbonnin** [comment]
> BTW. What about #2532? Is it the same problem?

Yep, #2532 is the same thing with the addition that we'd need a way to notify all listeners at once which we don't have at the moment.

Also, we're thinking about adding eviction APIs to the cache. Is it something you could potentially use or is your eviction logic custom so that it needs to be handled outside Apollo?

**Comment by ahetawal-p** [comment]
@martinbonnin I am also looking for a similar feature, as we need to clear some sensitive data from the the cache due to GDPR compliance, so if watcher some of these cleared records our UI needs to update in all places. 
And workaround which we can do for now ?
